### ISSUE TITLE
Urgent: Seeking New Developers and Maintainers

### ISSUE THREAD

**Issue by electriquo** [stoically/firefox-add-on-temporary-containers#634]
Dear Temporary Containers Community,

[It is with great sadness that the developer and the maintainer of the project, @stoically, passed away](#618).

The Temporary Containers extension has been an **invaluable tool for many users**, providing enhanced privacy and security while browsing the web. In light of the recent events, we are reaching out to the community to discuss the future of the project and find a suitable individual or group to take over its maintenance.

If you are interested in contributing to the continued development of Temporary Containers or know someone who might be a good fit for this role, please join the discussion here. I believe that by coming together as a community, we can ensure the longevity and improvement of this essential tool.

Your input and involvement in this process are crucial for the future of Temporary Containers. Let's work together to ensure that this valuable resource continues to thrive.

Thank you for your support.

**Comment by Gitoffthelawn** [comment]
Thank you for posting this.

I am willing to help, but alas, my available time is currently limited.  As such, I can see my role limited to:

1. Management.
2. Planning.
3. Ensuring that the code remain free of malware/spyware.
4. UI/UX design.
5. UI/UX top layer coding, including HTML and CSS code.
6. Bug reporting as I encounter issues.

What I, unfortunately, do not have sufficient time to do right now:
1. JavaScript coding.
2. Cross-platform and cross-browser testing.
3. Answering many questions from users.

**Comment by electriquo** [comment]
@Gitoffthelawn Thank you.
The call for a developer holds. We need a developer who can
- Update the code and its dependencies
- Fix issues
- Continue development, like developing features, handle deprecated parts, integrate browser/API changes, etc.

**Comment by dalto8** [comment]
Is there a list of priority issues anywhere that urgently need addressing?

I don't have the bandwidth to take over the entirety of the maintenance, but I would be willing to pitch in fix issues and contribute.

Also, does anyone have rights to merge code here or do we need to create a new fork?

**Comment by electriquo** [comment]
@dalto8

> Is there a list of priority issues anywhere that urgently need addressing?

No, nothing has been constructed since no one has admin/triage/maintain access in this repository.
Yet, I say that the fist thing we should do is to refresh the code by updating all dependencies/frameworks to the latest version. Is this this something you can do?

> does anyone have rights to merge code here or do we need to create a new fork?

Not that we know of. We've probably need to create a fork.

**Comment by tokariu** [comment]
it's 2024 and it's just now that I noticed the addon isn't working as expected anymore (like, domain exceptions not working anymore). it's the reason I looked up the addon status and it's shocking to read that the dev has passed away. my deep condolences to all the affected people.

I'm unfortunately unable to offer ressources to get this addon working again. But I want to emphasize that this addon is indeed invaluable for modern privacy oriented browsing. I hope this thread gets more attention to find new people willing to contribute to this addon. it took more than a year for me to notice the problem, there are probably many more not yet knowing what's going on.

**Comment by MurzNN** [comment]
As an alternative or a potential successor, I can recommend this extension: https://addons.mozilla.org/en-US/firefox/addon/containers-on-the-go/

**Comment by crssi** [comment]
> As an alternative or a potential successor, I can recommend this extension: https://addons.mozilla.org/en-US/firefox/addon/containers-on-the-go/

Why... nothing wrong with TemporaryContainers at the moment, and there is no extension getting even remotely close to this one.

**Comment by MurzNN** [comment]
> > As an alternative or a potential successor, I can recommend this extension: https://addons.mozilla.org/en-US/firefox/addon/containers-on-the-go/
> 
> Why... nothing wrong with TemporaryContainers at the moment, and there is no extension getting even remotely close to this one.

I totally agree with this, the Temporary Containers is the best!! But if it one day stops working for new versions of Firefox for some reason, and we will not have ways to fix this, we should have alternatives.

And probably it will be easier to port some cool features of the Temporary Containers to another addon, that is in active development, than trying to resume development of this extension, without actual write access to the main repo. Personally, I'm not happy with this way and prefer to reborn the Temporary Containers brand, but we should keep it in mind as an alternative.

**Comment by Gitoffthelawn** [comment]
> I totally agree with this, the Temporary Containers is the best!! But if it one day stops working for new versions of Firefox for some reason, and we will not have ways to fix this, we should have alternatives.
> 
> And probably it will be easier to port some cool features of the Temporary Containers to another addon, that is in active development, than trying to resume development of this extension, without actual write access to the main repo. Personally, I'm not happy with this way and prefer to reborn the Temporary Containers brand, but we should keep it in mind as an alternative.

I'm not sure Containers on the Go is in active development.  It's last update was in 2019, and it currently only has 300-400 telemetry-enabled installations.  That said, the `xpi` is less than 11Kb in size (which includes icons), so it does not have much code to maintain.

**Comment by anewuser** [comment]
I recommend this new lightweight extension to people who just want to be able to use disposable private tabs: https://addons.mozilla.org/addon/open-in-temp-container/

You need to go to the Firefox Add-ons Manager to see its options and make the toolbar button open new tabs. You can also give it permission to access your browsing history to prevent new URLs from being stored.

**Comment by filex100** [comment]
Is it possible to send an advertisement through the add-on?

Even if it's a minor update, but maybe that would help.


> I recommend this new lightweight extension to people who just want to be able to use disposable private tabs: https://addons.mozilla.org/addon/open-in-temp-container/
> 
> You need to go to the Firefox Add-ons Manager to see its options and make the toolbar button open new tabs. You can also give it permission to access your browsing history to prevent new URLs from being stored.

It doesn't look as comfortable as this one does.

If there is no alternative, maybe I won't use something similar and use other tools.

**Comment by electriquo** [comment]
> Is it possible to send an advertisement through the add-on?

Regardless whether this is a good thing to do, no one has the permissions to merge changes. A fork is a must.

**Comment by jonoff** [comment]
> > Is it possible to send an advertisement through the add-on?
> 
> Regardless whether this is a good thing to do, no one has the permissions to merge changes. A fork is a must.

A contributor could get around the need to fork as stated in this [comment](https://github.com/stoically/temporary-containers/issues/618#issuecomment-1688658808), and enacted the deceased user policy to transfer ownership

**Comment by boredsquirrel** [comment]
https://github.com/igorlogius/open-in-temp-container

This is the source code of the addon that is still in development. I already had a crash but otherwise it seems to have potential.

the code is very short and understandable.

the addon seems to delete the containers a minute after closing.

**Comment by VeryUsual** [comment]
Dear Temporary Containers Community,

I am deeply saddened by the loss of @stoically, the developer and maintainer of the Temporary Containers project. Their work has been crucial in providing enhanced privacy and security for many of us, and their absence is felt very deeply.

In light of these events, I wanted to share that I have taken the initiative to fork the Temporary Containers extension to ensure its continued development and support. You can find the forked repository here: https://github.com/temporary-containers-firefox/temporary-containers.

I am committed to maintaining and improving this valuable tool, but I welcome any contributions and collaboration from the community to help with this effort. If you have any suggestions, ideas, or if you're interested in contributing, please feel free to reach out through the repository or join the ongoing discussions.

Together, we can honor @stoically’s legacy by ensuring the continued success and enhancement of Temporary Containers. Thank you for your support and for being a part of this community.

Best regards,

ProgramminCat

**Comment by nord73** [comment]
Another path forward would be to entice Mozilla to include the feature set into core or MAC. There's at least this idea interested parties here might shake some life into: https://connect.mozilla.org/t5/ideas/integrate-features-of-quot-temporary-containers-quot-into/idi-p/16065.

**Comment by GodKratos** [comment]
I've started working on a fork to update this addon. Unfortunately since the modules haven't been updated in 5 years and are mostly all deprecated, I need to update all the dependencies and refactor everything to work with new versions of everything. This might take a while so if anyone has some experience with this and wants to help out, please let me know.

**Comment by dalto8** [comment]
I can provide a bit of time to help.

Maybe open issues on the fork so we can communicate?

**Comment by GodKratos** [comment]
> I can provide a bit of time to help.
> 
> Maybe open issues on the fork so we can communicate?

So far, I've just been working on updating the dependencies and getting Vue 3 working. I've committed what I've done so far (which has probably introduced a lot of problems that need to be fixed, but at least it is compiling now) here:
https://github.com/GodKratos/temporary-containers
Feel free to communicate there and I can open up access rights to collaborate as needed.

EDIT: I've moved the dependency upgrade to the "upgrade" branch to work on some fixes on main.

**Comment by GodKratos** [comment]
For anyone following this, a new version of the addon has been released here:
https://addons.mozilla.org/en-GB/firefox/addon/temporary-containers-plus/
I will be maintaining and updating that version from my fork of the project.

**Comment by electriquo** [comment]
@GodKratos Thank you for picking up the glove!
I've installed Temporary Container Plus. Once it will feel that everything is working, I will close this issue as you lead the way...

**Comment by Termuellinator** [comment]
@GodKratos And all future contributors: thank you so much! 👍
